deployment-server/src/Deploy/Worker.hs

```diff
@@ -0,0 +1,78 @@
+{-# LANGUAGE DeriveAnyClass             #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings          #-}
+module Deploy.Worker where
+
+import           Control.Concurrent.Chan     (Chan, readChan)
+import           Control.Monad               (forever)
+import           Control.Monad.IO.Class      (liftIO)
+import           Control.Monad.Reader        (runReaderT)
+import           Control.Newtype.Generics    (Newtype, unpack)
+import qualified Data.ByteString.Lazy.Char8  as BS
+import           Data.Char                   (isSpace)
+import           Data.Text                   (Text)
+import qualified Data.Text                   as Text
+import           GHC.Generics                (Generic)
+import           GitHub.Data.Webhooks.Events (PullRequestEvent)
+import           Options.Generic             (ParseField, ParseFields, ParseRecord, getRecord)
+import           System.Exit                 (ExitCode (ExitFailure, ExitSuccess))
+import           System.IO.Temp              (withSystemTempDirectory)
+import           System.Process.Typed        (readProcess, runProcess, setEnv, setWorkingDir, shell)
+import           Web.Slack                   (SlackConfig, chatPostMessage)
+import           Web.Slack.Chat              (mkPostMsgReq)
+
+newtype SlackChannel = SlackChannel Text
+    deriving stock (Generic)
+    deriving anyclass (Newtype, ParseFields, ParseRecord, ParseField)
+    deriving newtype (Show)
+
+instance Read SlackChannel where
+    readsPrec _ s = [(SlackChannel . Text.pack $ s, "")]
+
+newtype Deployment = Deployment Text
+    deriving stock (Generic)
+    deriving anyclass (Newtype, ParseFields, ParseRecord, ParseField)
+    deriving newtype (Show)
+
+instance Read Deployment where
+    readsPrec _ s = [(Deployment . Text.pack $ s, "")]
+
+runWorker :: Chan PullRequestEvent -> FilePath -> FilePath -> [String] -> SlackConfig -> SlackChannel -> Deployment -> IO ()
+runWorker chan configDir stateFile extraIncludes slackConfig slackChannel deployment = forever $ do
+    event <- readChan chan
+    deploy event configDir stateFile extraIncludes slackConfig slackChannel deployment
+
+deploy :: PullRequestEvent -> FilePath -> FilePath -> [String] -> SlackConfig -> SlackChannel -> Deployment -> IO ()
+deploy event configDir stateFile extraIncludes slackConfig slackChannel deployment = withSystemTempDirectory "deployment" $ \tempDir -> liftIO $ do
+    let plutusDir = tempDir <> "/plutus"
+        nixopsDir = plutusDir <> "/deployment/nixops"
+    putStrLn "Deploy origin/master"
+    runIn tempDir "git clone https://github.com/input-output-hk/plutus.git"
+    runIn plutusDir "git checkout origin/master"
+    (_, stdout', _) <- readIn plutusDir "git rev-parse HEAD"
+    let gitHead = rstrip . BS.unpack $ stdout'
```

Comment:
Is `gitHead` used?

Reply:
It was in a `putStrLn` so I've added it back in, also added to the slack alert.